cmd/ci-secret-bootstrap/main.go

```diff
@@ -503,12 +503,6 @@ func updateSecrets(getters map[string]Getter, secretsMap map[string][]*coreapi.S
 				}
 
 				if !shouldCreate {
-					for k, v := range existingSecret.Data {
-						if _, exists := secret.Data[k]; exists {
-							continue
-						}
-						secret.Data[k] = v
-					}
```

Comment:
Maybe I am paranoid, but can we be super-careful here to not destroy anything important before we merge this? Could we perhaps run this in some dry mode and inspect the list of records that this will delete on one of our clusters?

Reply:
It should be worth the try. I will check and I find any important keys that we shouldn't delete, it would be a mystery how they ended up there since they don't exist in any of the configurations.

Reply:
Could we do it like those?

Step 1:
Add a logging line before `secret.Data[k] = v` indicating that the key will be deleted (because it is not from Vault/Config).

Step 2:
After running the tool, checking the above logs. Remove `secret.Data[k] = v` after we can justify our deleting actions.

Reply:
Maybe this is a counterexample?

https://github.com/openshift/release/blob/3203fcb74a4a2ec0707b268783926c94ff43c2ce/core-services/ci-secret-bootstrap/_config.yaml#L420

Reply:
This one won't be generated as a key in any secret. Is there any other case that a key exists in a secret but it doesn't exist in Vault?

Reply:
That is all I can think of right now.
I will know it from the logging in Step 1.

Reply:
Yea I like the idea of logging the stale keys.

Reply:
But this is still not testable. I will try to find a way to get the list of the stale keys from our cluster. But.... if there is a key that doesn't exist in Vault and it shouldn't exist in any secret in any cluster, it doesn't worth the effort of finding them...